asteroid/filterbanks/multiphase_gammatone_filterbank.py

```diff
@@ -0,0 +1,108 @@
+import numpy as np
+import torch
+from .enc_dec import Filterbank
+
+class MultiphaseGammatoneFB(Filterbank):
+    """ Multi-Phase Gammatone Filterbank as described in [1].
+        Please cite [1] whenever using this.
+        Original code repository: `<https://github.com/sp-uhh/mp-gtf>`
+
+    References:
+    [1] David Ditter, Timo Gerkmann, "A Multi-Phase Gammatone Filterbank for Speech
+        Separation via TasNet", IEEE Int. Conf. Acoust., Speech, Signal Process. (ICASSP),
+        Barcelona, Spain, May 2020.
+        Available: `<https://ieeexplore.ieee.org/document/9053602/>`
+
+    Args:
+        n_filters (int): Number of filters. 
+        kernel_size (int): Length of the filters.
+        sample_rate (int, optional): The sample rate (used for initialization).
+        stride (int, optional): Stride of the convolution. If None (default),
+            set to ``kernel_size // 2``.
+    """
+    
+    def __init__(self, n_filters=128, kernel_size=16, sample_rate=8000, stride=None):
+        super(MultiphaseGammatoneFB, self).__init__(n_filters, kernel_size, stride=stride)
+        self.sample_rate = sample_rate
+        self.n_filters = n_filters
+        self.kernel_size = kernel_size
+
+    @property
+    def filters(self):
+        """ Compute filters from parameters """
+        length_in_seconds = self.kernel_size / self.sample_rate
+        filterbank = generate_mpgtf(self.sample_rate, length_in_seconds, self.n_filters)
+        filterbank = np.expand_dims(filterbank, axis=1)
+        return torch.from_numpy(filterbank).float()
```

Comment:
This function is always called on forward. This means the filters will be generated everytime. That's not possible. 
The filters property is to enable dynamically generated filters. 
MPGTF are static, check out the implementations of the STFT and Free filterbank (which are also static) 
You'll have to register a buffer (to automatize device transfer and save with the state_dict), `_filters` and the function `filters` will return this buffer `_filters`. 
The buffer `_filters` is created in the init. 

Tell me if this is clear.

Reply:
That makes total sense. Changed this such that now it is only generated once.